packages/backend-api/src/Model/Product/ProductApiDataValidator.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\BackendApiBundle\Model\Product;
+
+use DateTime;
+use Shopsys\BackendApiBundle\Component\Validator\AbstractValidator;
+use Shopsys\FrameworkBundle\Component\Domain\Domain;
+use Symfony\Component\Validator\Constraints;
+
+class ProductApiDataValidator extends AbstractValidator implements ProductApiDataValidatorInterface
```

Comment:
The `AbstractValidator` shouldn't be abstract and should be kind of `ValidatorRunner` with public method. And should be used here as private `$validatorRunner`